### PR TITLE
feat: new configuration UI for API Explorer

### PR DESCRIPTION
**Pull request by jkaster**
- removed `ConfigDialog`
- moved `CodeCopy` to `@looker/code-editor`
- simplified `LoginForm`
- added **Configure** tab to RunIt
- upgraded to looker/components 2.x, and other related packages
- TODO `SideNav` needs to be upgraded to `Accordion2`
- TODO e2e test `renders a tag scene and filters by operation` is failing

### CODE REVIEW

File: packages/run-it/src/components/common/common.tsx
Comment (bryans99) [inline packages/run-it/src/components/common/common.tsx:27]: I'm looking at the components in this file and i'm not seeing a lot of meat. Not totally convinced but I can sort of see it. I vaguely remember that defaultProps caused problems for tree shaking so this will probably need to be redone. 

I see why its in run it but maybe its time to create an "sdk-components" package (for want of a better name).

Not saying part if this PR. Just a general observation.

File: packages/api-explorer/src/components/common/index.ts
Comment (bryans99) [inline packages/api-explorer/src/components/common/index.ts:27]: This is fine but I think the common directory kind of sucks. I think the two components that remain should be promoted to components.

Not part of this PR though.

File: packages/run-it/src/RunIt.tsx
Comment (bryans99) [inline packages/run-it/src/RunIt.tsx:167]: is preventDefault needed?

Comment (bryans99) [reply packages/run-it/src/RunIt.tsx:167]: It's need for form submit, I dont think its needed for button click (i believe this is what it associated with).

For handleSubmit I would call preventDefault in the RequestForm component.

Again, change not needed for this PR.

File: packages/run-it/src/RunIt.tsx
Comment (bryans99) [inline packages/run-it/src/RunIt.tsx:274]: As an aside, Lara Harrow was showing me Tabs2 component today. Much easier to use.

File: packages/run-it/src/components/ConfigForm/ConfigForm.tsx
Comment (bryans99) [inline packages/run-it/src/components/ConfigForm/ConfigForm.tsx:138]: is updateMessage a better name?

**Comment by bryans99** [comment]
Probably unrelated to this PR but it would be nice to add tooltip to clear button. It's not obvious if there is a relationship with the login button

image

**Comment by bryans99** [comment]
I clicked verify here

image


And this was the result


image

The message that OAUTH is configured seems to be incorrect.

image

Not seeing it in local storage

image

I clicked login and it went off and came back so it must have had it in memory but not in local storage. On return looks like this

image

**Review by bryans99 — COMMENTED** [review]
Code is okay but I think we need to open up some tickets for future fixits.

There was some issues with testing however. I can show you in the morning

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (jkaster) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:79]: I think I'll try to get it done for this PR tomorrow. I've budgeted 2 more days for this PR.

File: packages/run-it/src/components/common/common.tsx
Comment (jkaster) [reply packages/run-it/src/components/common/common.tsx:27]: Yeah, definitely worth considering.

File: packages/api-explorer/src/components/common/index.ts
Comment (jkaster) [reply packages/api-explorer/src/components/common/index.ts:27]: Thanks. I may get to that for this PR. If not, we should have a refactor PR soon.

File: packages/run-it/src/RunIt.tsx
Comment (jkaster) [reply packages/run-it/src/RunIt.tsx:274]: Oh cool. I'll upgrade to that tomorrow.

File: packages/run-it/src/components/ConfigForm/ConfigForm.tsx
Comment (jkaster) [reply packages/run-it/src/components/ConfigForm/ConfigForm.tsx:138]: Yes it absolutely is. I'll make that change tomorrow too.

**Comment by jkaster** [comment]
I'll catch up with you about this tomorrow. As I mentioned elsewhere, I've got 2 more days budgeted for this PR so I can definitely do more work to improve it before we merge it. We may make it part of another PR but there's also going to be an issue going from a DevPortal spec key to a Looker server spec key that we'll need to handle before this can be deployed to the DevPortal.

File: packages/run-it/src/RunIt.tsx
Comment (jkaster) [reply packages/run-it/src/RunIt.tsx:167]: Definitely worth testing these overrides to see if they're needed. Is it just to prevent the "submit" action? I'll find out tomorrow.

File: packages/run-it/src/RunIt.tsx
Comment (jkaster) [reply packages/run-it/src/RunIt.tsx:274]: `Tabs2` isn't quite ready yet. I'll create a ticket for that.

File: packages/run-it/src/components/common/common.tsx
Comment (jkaster) [reply packages/run-it/src/components/common/common.tsx:27]: This becomes even more valuable when we lazy load the various response renderers

File: packages/api-explorer/jest-puppeteer.config.js
Comment (bryans99) [inline packages/api-explorer/jest-puppeteer.config.js:40]: Delete commented out code?

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (bryans99) [inline packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:87]: Ticket?

File: packages/run-it/src/RunIt.tsx
Comment (bryans99) [inline packages/run-it/src/RunIt.tsx:271]: Get rid of the ternary?
!isExtension instead?
The current true side is <></> which is nothing

**Review by bryans99 — COMMENTED** [review]
Behavior much better but I personally find it a little jaring when save collapses the accordion. I get it but I find it weird. I also know its not your design.

Sidebar looks off. No spacing or margin,

Will approve if you think your ready

**Comment by lukelooker** [comment]
Removed myself as a reviewer since I contributed code / am no longer unbiased :)

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (jkaster) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx]: Migrated

File: packages/api-explorer/jest-puppeteer.config.js
Comment (jkaster) [reply packages/api-explorer/jest-puppeteer.config.js:40]: Thanks for catching. I needed to revert my debugging changes. @josephaxisa and I discussed having a "debug mode" for puppeteer also

File: packages/run-it/src/RunIt.tsx
Comment (jkaster) [reply packages/run-it/src/RunIt.tsx:271]: When I try to do that there are syntax errors for `Tab` and `TabPanels`. Leaving as is.

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (jkaster) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx]: Created

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (bryans99) [inline packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:68]: Gonna assume ticket for this

File: packages/api-explorer/src/scenes/MethodScene/MethodScene.tsx
Comment (bryans99) [inline packages/api-explorer/src/scenes/MethodScene/MethodScene.tsx:101]: in theory should include setOn but I think that lint rule is kind of stupid

File: packages/code-editor/README.md
Comment (bryans99) [inline packages/code-editor/README.md:69]: I think this should be CodeCopy.

File: packages/code-editor/README.md
Comment (bryans99) [inline packages/code-editor/README.md:71]: This component wraps instead of extends? Being an OO purist here

**Review by bryans99 — CHANGES_REQUESTED** [review]
Looks good overall. Requesting changes to the code editor readme as this is blocked waiting on Jared's cla approval anyway. Will approve if you want me too.

There is the issue with switching back to 3.1 after logging in with 4.0. In my opinion this PR is too big so would prefer a separate PR. However, as this PR is blocked I don't mind reviewing that change as part of this PR.

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx
Comment (jkaster) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:68]: Already created

File: packages/code-editor/README.md
Comment (jkaster) [reply packages/code-editor/README.md]: corrected

File: packages/code-editor/README.md
Comment (jkaster) [reply packages/code-editor/README.md]: Using "wraps" now

File: packages/code-editor/README.md
Comment (bryans99) [inline packages/code-editor/README.md:43]: extends or wraps in this scenario? props extends, component wraps :D . Went back and looked at CodeCopy and its the same thing. My gut feeling is that technically it should be wraps but I can live with extends because of the props.

Gonna approve anyway

**Review by bryans99 — APPROVED** [review]
I found an inconsistency in the readme, caused by my earlier comment. I can go with extends or wraps now.

File: packages/code-editor/README.md
Comment (jkaster) [reply packages/code-editor/README.md:43]: Not gonna change this one right now. We can update when we figure out how to support the copy clipboard button better than `CodeCopy` does

Comment (jkaster) [reply packages/code-editor/README.md:43]: Just updated this because I needed to revert `clipboard` in types.ts

**Review by bryans99 — APPROVED** [review]
LGTM

**Comment by jhardy** [comment]
@googlebot I consent.

**Pull request merged**
